doc/mgmt/SONiC_Ztp_DesignDoc.md

```diff
@@ -0,0 +1,254 @@
+
+Note: The above text is taken from the sonic-ztp readme (https://github.com/Azure/SONiC/blob/master/doc/ztp/ztp.md)
+
+## 2.2 Functional Description
+
+Upon receiving a request, the app module forwards the request to the service running at the host, which calls the backend implementation to do all ztp related tasks.
+
+# 3 Design
+## 3.1 Overview
+TBD
+
+## 3.2 DB Changes
+
+N/A
+### 3.2.1 CONFIG DB
+
+N/A
+
+### 3.2.2 APP DB
+
+N/A
+
+### 3.2.3 STATE DB
+
+TBD. 
+
+### 3.2.4 ASIC DB
+
+N/A
+
+### 3.2.5 COUNTER DB
+
+N/A
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+
+N/A
+
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+TBD (Working on a custom Sonic yang).
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+
+**ztp enable**
+
+```
+sonic# ztp enable
+Success
+```
+
+**ztp disable**
+
+```
+sonic# ztp disable
+Success
+```
+
+**ztp run**
+
+```
+sonic# ztp run
+Success
+```
+
+#### 3.6.2.2 Show Commands
+
+**sonic_installer list**
+
+```
+sonic# show ztp-status
+ZTP Admin Mode : True
+ZTP Service    : Inactive
+ZTP Status     : SUCCESS
+ZTP Source     : dhcp-opt67 (eth0)
+Runtime        : 05m 31s
+Timestamp      : 2019-09-11 19:12:24 UTC
+
+ZTP Service is not running
+
+01-configdb-json: SUCCESS
+02-connectivity-check: SUCCESS
+```
+
+Use the verbose option to display more detailed information.
+
+```
+sonic# show ztp-status --verbose
+========================================
+ZTP
+========================================
+ZTP Admin Mode : True
+ZTP Service    : Inactive
+ZTP Status     : SUCCESS
+ZTP Source     : dhcp-opt67 (eth0)
+Runtime        : 05m 31s
+Timestamp      : 2019-09-11 19:12:16 UTC
+ZTP JSON Version : 1.0
+
+ZTP Service is not running
+
+----------------------------------------
+01-configdb-json
+----------------------------------------
+Status          : SUCCESS
+Runtime         : 02m 48s
+Timestamp       : 2019-09-11 19:11:55 UTC
+Exit Code       : 0
+Ignore Result   : False
+
+----------------------------------------
+02-connectivity-check
+----------------------------------------
+Status          : SUCCESS
+Runtime         : 04s
+Timestamp       : 2019-09-11 19:12:16 UTC
+Exit Code       : 0
+Ignore Result   : False
+```
+
+Note: The above text is taken from the sonic-ztp readme (https://github.com/Azure/SONiC/blob/master/doc/ztp/ztp.md)
+
+#### 3.6.2.3 Debug Commands
+
+N/A
+
+#### 3.6.2.4 IS-CLI Compliance
+
+N/A
+
+### 3.6.3 REST API Support
+
+**TODO**
+TBD (Working on a custom Sonic yang).
+
+# 4 Flow Diagrams
+N/A
+
+# 5 Error Handling
+
+TBD
+
+# 6 Serviceability and Debug
+
+TBD
+
+# 7 Warm Boot Support
+
+TBD
+
+# 8 Scalability
+N/A
+
+# 9 Unit Test
+List unit test cases added for this feature including warm boot.
```

Comment:
Need UT cases here in table format. 
| Test Name | Test Description |
|---|---|
| Toggle ZTP | Toggle ZTP on and off and check status |
| Test2 | Description2 |